src/components/external-source/ExternalSourceManager.svelte

```diff
@@ -0,0 +1,897 @@
+      }
+    }
+  }
+
+  // Column definition
+  $: columnDefs = [
+    ...baseColumnDefs,
+
+    {
+      field: 'valid_at',
+      filter: 'text',
+      headerName: `Valid At (${$plugins.time.primary.label})`,
+      resizable: true,
+      sort: 'desc',
+      sortable: true,
+      valueGetter: (params: ValueGetterParams<ExternalSourceSlim>) => {
+        if (params.data?.valid_at) {
+          return formatDate(new Date(params.data?.valid_at), $plugins.time.primary.format);
+        }
+      },
+    },
+    {
+      field: 'start_time',
+      filter: 'text',
+      headerName: `Start Time (${$plugins.time.primary.label})`,
+      resizable: true,
+      sortable: true,
+      valueGetter: (params: ValueGetterParams<ExternalSourceSlim>) => {
+        if (params.data?.start_time) {
+          return formatDate(new Date(params.data?.start_time), $plugins.time.primary.format);
+        }
+      },
+    },
+    {
+      field: 'end_time',
+      filter: 'text',
+      headerName: `End Time (${$plugins.time.primary.label})`,
+      resizable: true,
+      sortable: true,
+      valueGetter: (params: ValueGetterParams<ExternalSourceSlim>) => {
+        if (params.data?.end_time) {
+          return formatDate(new Date(params.data?.end_time), $plugins.time.primary.format);
+        }
+      },
+    },
+    {
+      field: 'created_at',
+      filter: 'text',
+      headerName: `Created At (${$plugins.time.primary.label})`,
+      resizable: true,
+      sortable: true,
+      valueGetter: (params: ValueGetterParams<ExternalSourceSlim>) => {
+        if (params.data?.valid_at) {
+          return formatDate(new Date(params.data?.created_at), $plugins.time.primary.format);
+        }
+      },
+    },
+    {
+      cellClass: 'action-cell-container',
+      cellRenderer: (params: SourceCellRendererParams) => {
+        const actionsDiv = document.createElement('div');
+        actionsDiv.className = 'actions-cell';
+        new DataGridActions({
+          props: {
+            deleteCallback: data => params.onDeleteExternalSource([data]),
+            deleteTooltip: {
+              content: 'Delete External Source',
+              placement: 'bottom',
+            },
+            hasDeletePermission: hasDeleteExternalSourcePermissionOnRow(user, params.data),
+            rowData: params.data,
+          },
+          target: actionsDiv,
+        });
+
+        return actionsDiv;
+      },
+      cellRendererParams: {
+        onDeleteExternalSource,
+      } as CellRendererParams,
+      field: 'actions',
+      headerName: '',
+      resizable: false,
+      sortable: false,
+      suppressAutoSize: true,
+      suppressSizeToFit: true,
+      width: 25,
+    },
+  ];
+
+  // Selected elements and values
+  $: effects
+    .getExternalEvents(
+      selectedSource ? selectedSource.key : null,
+      selectedSource ? selectedSource.derivation_group_name : null,
+      user,
+    )
+    .then(
```

Comment:
I think this can just be reduced to 
```    .then(fetched => (selectedEvents = fetched));```
The time conversions are already happening in the `effects
    .getExternalEvents` function